libcoap-sys/build.rs

```diff
@@ -141,27 +141,57 @@ fn main() {
                             ),
                         );
                     };
+
+                    if let Some(tinydtls_libs) = env::var_os("DEP_TINYDTLS_LIBS") {
+                        build_config.env("TinyDTLS_LIBS", format!("-L{}", tinydtls_libs.to_str().unwrap()));
+
+                        build_config.env(
+                            "PKG_CONFIG_PATH",
+                            Path::new(tinydtls_libs.as_os_str())
+                                .join("lib")
+                                .join("pkgconfig")
+                                .into_os_string(),
+                        );
+                    }
                 },
                 DtlsBackend::OpenSsl => {
                     // Set include path according to the path provided by openssl-sys (relevant if
                     // openssl-sys is vendored)
                     if let Some(openssl_include) = env::var_os("DEP_OPENSSL_INCLUDE") {
                         build_config.env("OpenSSL_CFLAGS", format!("-I{}", openssl_include.to_str().unwrap()));
+                        build_config.env(
+                            "PKG_CONFIG_PATH",
+                            Path::new(openssl_include.as_os_str())
+                                .parent()
+                                .unwrap()
+                                .join("lib")
+                                .join("pkgconfig")
+                                .into_os_string(),
+                        );
                     }
                 },
                 DtlsBackend::MbedTls => {
                     // Set include path according to the path provided by mbedtls-sys (relevant if
                     // mbedtls-sys is vendored).
-                    // libcoap doesn't support overriding the MbedTLS CFLAGS, but doesnt set those
+                    // libcoap doesn't support overriding the MbedTLS CFLAGS, but doesn't set those
                     // either, so we just set CFLAGS and hope they propagate.
                     if let Some(mbedtls_include) = env::var_os("DEP_MBEDTLS_INCLUDE") {
                         build_config.env("CFLAGS", format!("-I{}", mbedtls_include.to_str().unwrap()));
+                        build_config.env(
+                            "PKG_CONFIG_PATH",
+                            Path::new(mbedtls_include.as_os_str())
+                                .parent()
+                                .unwrap()
+                                .join("lib")
+                                .join("pkgconfig")
+                                .into_os_string(),
```

Comment:
Maybe this is overeager generalization, but since this chain occurs in the code three times, it could be moved into a function.

Reply:
I think that generalizing this would make more work than just keeping it as is.